File: ultimate/AuthenticationMiddlewareHandler.java
package ultimate;

import chainofresponsibility.AbstractMiddlewareHandler;

public class AuthenticationMiddlewareHandler extends AbstractMiddlewareHandler {

    @Override
    public MiddlewareResponse handle(HttpRequest request) {
        if (request.getUser().isAuthenticated()) {
            super.handle(request);
        }
        return new MiddlewareResponse("Request has failed to be authenticated", false);
    }
}

File: typescript/patterns/ultimate-design-pattern-challenge/authentication-middleware-handler.ts
class AuthenticationMiddlewareHandler extends AbstractMiddlewareHandler {
  override handle(request: HttpRequest): MiddlewareResponse {
    if (!request.getUser().getIsAuthenticated()) {
      return new MiddlewareResponse(
        "Request has failed to be authenticated",
        false,
      );
    }

    return super.handle(request);
  }
}
